### PR TITLE
Add the ability for CssBuilder to properly Add multiple classes without duplicates

### PR DESCRIPTION
**Pull request by StevenTCramer**
# Pull Request

## 📖 Description

1. **Adding AddClasses Method:**
   - Introduced `AddClasses` to handle multiple space-separated CSS classes efficiently, leveraging the `HashSet` to eliminate duplicates.

2. **Refactoring AddClass Methods:**
   - Updated `AddClass` methods to internally call `AddClasses`, ensuring backward compatibility.

3. **Conditional AddClasses Methods:**
   - Added conditional `AddClasses` methods to add classes based on boolean conditions or functions.

These changes optimize class addition by preventing duplicates and improving performance.

### 🎫 Issues

The current AddClass method actually allows for adding multiple but it doesn't split the input and utilize the hashset and thus would allow duplicates.

## 📑 Test Plan

Update CSSBuilder Tests cases to cover 

### General

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fluentui-blazor/blob/master/docs/contributing.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have added [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for my new compontent
- [ ] I have modified an existing component
- [ ] I have validated the [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for an existing component

### CODE REVIEW

**Review by vnbaaij — APPROVED** [review]

**Comment by dvoituron** [comment]
This could be an interesting improvement. But why not keep the existing ˋAddClass` method, improving the internal code? And don't add new methods that don't solve the current usage.

The CSS class name must be composed of numbers, hyphens, underscores and letters.

So the existing ˋAddClass` method could be split into spaces, commas, ... to use your code.

**Comment by StevenTCramer** [comment]
* Update AddClass to support single or multiple classes.  
* Ignore invalid classes based on the regex. 
* Updated tests cases to validate.
*  Fixed StyleBuilder Tests copy and paste errors.

File: tests/Core/Utilities/StyleBuilderTests.cs [hunk 1]
```diff
@@ -71,20 +71,6 @@ public void CssBuilder_WithComplexUserStyle()
         Assert.Equal("color: red; font-size: 12px; font-name: courier;", styleBuilder.Build());
     }
 
-    [Fact]
```
Comment (dvoituron) [inline tests/Core/Utilities/StyleBuilderTests.cs:74]: You can perhaps keep (and correct as follows) this (additional) test to check the use of several classes.

```csharp
[Fact]
public void CssBuilder_WithComplexUserClasses()
{
    // Assert
    var cssBuilder = new CssBuilder("   my-user-class1     my-user-class2  ");

    // Act
    cssBuilder.AddClass("  class1  ");
    cssBuilder.AddClass("  class2  ");

    // Assert - Values are sorted
    Assert.Equal("class1 class2 my-user-class1 my-user-class2", cssBuilder.Build());
}
```

**Review by dvoituron — CHANGES_REQUESTED** [review]
Good work. I like it :-)

File: tests/Core/Utilities/StyleBuilderTests.cs (hunk 1 above)
Comment (StevenTCramer) [reply tests/Core/Utilities/StyleBuilderTests.cs:74]: That was a CssBuilder test in the `StyleBuilderTests.cs` file.  And we have similar tests

CssBuilder_AddMultipleValidUserClasses
CssBuilder_AddClassesWithExtraSpaces

In the appropriate CssBuilderTests.cs file.

But I did see I was missing a few  test cases.  I added.

CssBuilder_AddMultipleClassesAtOnce
CssBuilder_AddMultipleClassesAtOnceWithUserClass
CssBuilder_AddMultipleClassesAtOnceWithExtraSpaces

**Comment by azunyuuuuuuu** [comment]
This commit sadly breaks using Tailwind CSS with this library. Classes like `hover:bg-red-400`, `grid-cols-[1fr_3fr]` or `[&>a]:text-sm` get eaten up by the regex.

Is there a way to disable or change the behaviour of the validation?

**Comment by vnbaaij** [comment]
No, currently there is no way to prevent that from happening.

**Comment by dvoituron** [comment]
> This commit sadly breaks using Tailwind CSS with this library. Classes like `hover:bg-red-400`, `grid-cols-[1fr_3fr]` or `[&>a]:text-sm` get eaten up by the regex.

Weird. These strings are not valid class names.

> a name must begin with an underscore (_), a hyphen (-), or a letter(a–z), followed by any number of hyphens, underscores, letters, or numbers.
https://stackoverflow.com/questions/448981/which-characters-are-valid-in-css-class-names-selectors

**Pull request merged**